lib/ansible/modules/windows/win_file.py

```diff
@@ -28,32 +28,53 @@
-       and can create or remove directories.
-     - Unlike M(file), does not modify ownership, permissions or manipulate links.
+  - Sets attributes of files, symlinks, hard links, junction points and directory.
+  - Can create files, symlinks, hard links, junction points, directories and files.
+  - Unlike M(file), does not modify ownership or permissions.
 notes:
-    - See also M(win_copy), M(win_template), M(copy), M(template), M(assemble)
-requirements: [ ]
+  - See also M(win_copy), M(win_template), M(copy), M(template), M(assemble)
 author: "Jon Hawkesworth (@jhawkesworth)"
 options:
+  force:
+    description:
+      - This flag indicates whether to force the creation of a link in the
+        following cases.
+      - If C(symbolic_link) and C(path) exists and is not a symbolic link, the
+        C(path) will be deleted and set as a new symbolic link.
+      - If C(junction_point) and C(path) exists and is not a junction point,
+        the C(path) will be deleted and set as a new junction point.
+      - If C(hard_link) and C(path) exists and is not a hard link of C(src),
+        the C(path) will be deleted and set as a new hard link.
+    choices: [true, false]
+    version_added: "2.4"
   path:
     description:
-      - 'path to the file being managed.  Aliases: I(dest), I(name)'
+      - 'path to the file being managed. Aliases: I(dest), I(name)'
     required: true
     aliases: ['dest', 'name']
+  src:
+    description:
+      - Path of the file to link to (applies only to
+        C(state=symbolic_lint/junction_point/hard_link)).
+      - If C(state=junction_point) this needs to be a directory.
+      - If C(state=hard_link) this needs to be a file.
+    version_added: "2.4"
   state:
     description:
       - If C(directory), all immediate subdirectories will be created if they
         do not exist.
-        If C(file), the file will NOT be created if it does not exist, see the M(copy)
+      - If C(file), the file will NOT be created if it does not exist, see the M(copy)
         or M(template) module if you want that behavior.  If C(absent),
         directories will be recursively deleted, and files will be removed.
-        If C(touch), an empty file will be created if the C(path) does not
-        exist, while an existing file or directory will receive updated file access and
+      - If C(touch), an empty file will be created if the C(path) does not
+        exist, while an existing file or directory will receive updated
         modification times (similar to the way C(touch) works from the command line).
-    choices: [ file, directory, touch, absent ]
+      - If C(symbolic_link) a symbolic link (soft link) will be created or changed. Added in 2.4.
+      - If C(junction_point) a junction point will be created or changed. Added in 2.4.
+      - If C(hard_link) a hard link will be created. Added in 2.4.
+    choices: [ file, directory, touch, absent, symbolic_link, junction_point, hard_link ]
```

Comment:
Just a matter of opinion, I don't like the strings symbolic_link, junction_point and hard_link.
I would prefer: *symlink*, *hardlink* and *something* else for junction_point :-)

The strings for Linux/Unix is *link* and *hard* which is also not optimal ;-)